In [1]:
import pandas as pd

**Pregunta1**

¿cuál fue el rendimiento total en los tres años de la inversión?

In [5]:
tabla = pd.DataFrame(data={'año':[1, 2, 3], 'rend':[0.2, -0.1, 0.15]})
tabla['1+rend'] = 1 + tabla['rend']
tabla['acum'] = tabla['1+rend'].cumprod()
tabla

año  rend  1+rend   acum
0    1  0.20    1.20  1.200
1    2 -0.10    0.90  1.080
2    3  0.15    1.15  1.242

In [6]:
tabla.iloc[-1, -1] - 1

0.242

**Respuesta**

$24.2%$

**Pregunta 2**

Suponga que las probabilidades de recesión, normalidad y expansión en la economía son de 40%, 30%, y 30% respectivamente. Además, suponga que va a tener un rendimiento anual en sus inversiones del -10% en caso de recesión, 8% en caso de normalidad y 23% en caso de expansión. ¿Cuál sería el rendimiento esperado anual de sus inversiones?

In [7]:
tabla2 = pd.DataFrame(data={'prob':[0.4, 0.3, 0.3]
                            ,'rend':[-0.1, 0.08, 0.23]})
tabla2

prob  rend
0   0.4 -0.10
1   0.3  0.08
2   0.3  0.23

In [8]:
#rendimiento esperado anual
Er = (tabla2['prob'] * tabla2['rend']).sum()
Er

0.053

**Respuesta**

$5.3 %$

**Pregunta 3**

Suponga que las probabilidades de recesión, normalidad y expansión en la economía son de 40%, 30%, y 30% respectivamente. Además, suponga que va a tener un rendimiento anual en sus inversiones del -10% en caso de recesión, 8% en caso de normalidad y 23% en caso de expansión. ¿Cuál sería el rendimiento esperado anual de sus inversiones?

In [9]:
tabla2

prob  rend
0   0.4 -0.10
1   0.3  0.08
2   0.3  0.23

In [12]:
sr2 = (tabla2['prob'] * (tabla2['rend'] - Er)**2).sum()
sr2**0.5

0.13777155003846042